cpu/nrf52/radio/nrf802154/nrf802154.c

```diff
@@ -120,6 +143,63 @@ static void _enable_tx(void)
     DEBUG("[nrf802154] Device state: TXIDLE\n");
 }
 
+/**
+ * @brief   Convert from dBm to the internal representation, when the
+ *          radio operates as a IEEE802.15.4 transceiver.
+ */
+static inline uint8_t _dbm_to_ieee802154_hwval(int8_t dbm)
+{
+    return ((dbm - NRF802154_ED_RSSIOFFS) / NRF802154_ED_RSSISCALE);
+}
+
+/**
+ * @brief   Convert from the internal representation to dBm, when the
+ *          radio operates as a IEEE802.15.4 transceiver.
+ */
+static inline int8_t _hwval_to_ieee802154_dbm(uint8_t hwval)
+{
+    return NRF802154_ED_RSSIOFFS + NRF802154_ED_RSSISCALE * hwval;
+}
+
+/**
+ * @brief   Get CCA threshold value in internal represetion
+ */
+static int _get_cca_thresh(void)
+{
+    return (NRF_RADIO->CCACTRL & RADIO_CCACTRL_CCAEDTHRES_Msk) >>
+            RADIO_CCACTRL_CCAEDTHRES_Pos;
+}
+
+/**
+ * @brief   Set CCA threshold value in internal represetion
+ */
+static void _set_cca_thresh(uint8_t thresh)
+{
+    NRF_RADIO->CCACTRL &= ~RADIO_CCACTRL_CCAEDTHRES_Msk;
+    NRF_RADIO->CCACTRL |= thresh << RADIO_CCACTRL_CCAEDTHRES_Pos;
+}
+
+/**
+ * @brief   Check wether the channel is clear or not
+ * @note    So far only CCA with Energy Detection is supported (CCA_MODE=1).
+ */
+static bool _channel_is_clear(void)
+{
+    NRF_RADIO->CCACTRL |= RADIO_CCACTRL_CCAMODE_EdMode;
+    NRF_RADIO->EVENTS_CCAIDLE = 0;
+    NRF_RADIO->EVENTS_CCABUSY = 0;
+    NRF_RADIO->TASKS_CCASTART = 1;
+
+    for(;;) {
```

Comment:
Just out of interest, did you check if you ever mad a second operation here?

Reply:
here there's only one or the other, so either one returning something is valid

Reply:
or you mean something else?

Reply:
I meant iteration